src/backend/distributed/commands/schema_based_sharding.c

```diff
@@ -168,3 +183,238 @@ RegisterTenantSchema(Oid schemaId)
 
 	return TenantSchemaInsertCommand(schemaId, colocationId);
 }
+
+
+/*
+ * UnRegisterTenantSchema unregisters given tenant schema back to regular schema
+ * locally and returns the command to do the same on the workers.
+ */
+char *
+UnRegisterTenantSchema(Oid schemaId)
+{
+	CheckCitusVersion(ERROR);
+
+	uint32 colocationId = SchemaIdGetTenantColocationId(schemaId);
+	if (colocationId == INVALID_COLOCATION_ID)
+	{
+		ereport(ERROR, (errmsg("schema \"%s\" is not a tenant schema",
+							   get_namespace_name(schemaId))));
+	}
+
+	DeleteTenantSchemaLocally(schemaId);
+	DeleteColocationGroupLocally(colocationId);
+
+	return TenantSchemaDeleteCommand(schemaId);
+}
+
+
+/*
+ * GetRelationIdsInSchema returns all nonshard relation ids, with relkind = relation or
+ * partitioned, inside given schema.
+ */
+static List *
+GetRelationIdsInSchema(Oid schemaId)
+{
+	List *relationIdList = NIL;
+
+	/* scan all relations in pg_class and return all tables inside given schema */
+	Relation relationRelation = relation_open(RelationRelationId, RowExclusiveLock);
+
+	ScanKeyData scanKey[1] = { 0 };
+	ScanKeyInit(&scanKey[0], Anum_pg_class_relnamespace, BTEqualStrategyNumber,
+				F_OIDEQ, ObjectIdGetDatum(schemaId));
+	SysScanDesc scanDescriptor = systable_beginscan(relationRelation, ClassNameNspIndexId,
+													false, NULL, 1, scanKey);
+
+	HeapTuple heapTuple = NULL;
+	while ((heapTuple = systable_getnext(scanDescriptor)) != NULL)
+	{
+		Form_pg_class relationForm = (Form_pg_class) GETSTRUCT(heapTuple);
+		char *relationName = NameStr(relationForm->relname);
+		Oid relationId = get_relname_relid(relationName, schemaId);
+
+		if (!OidIsValid(relationId))
+		{
+			continue;
+		}
+
+		/* skip shards */
+		if (RelationIsAKnownShard(relationId))
+		{
+			continue;
+		}
+
+		/* skip partitions, they would be distributed by partitioned table */
+		if (PartitionTable(relationId))
+		{
+			continue;
+		}
+
+		/*
+		 * Collect only the relkinds of relation and partitioned since others are
+		 * distributed on workers.
+		 */
+		if (RegularTable(relationId))
+		{
+			relationIdList = lappend_oid(relationIdList, relationId);
+		}
```

Comment:
or should we not skip foreign tables here and let CreateTenantSchemaTable to throw an error later? (cc: @marcocitus)